pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2154,8 +2154,13 @@ public boolean isSingleTenantCluster() {
       ZNRecord record = helixInstanceConfig.getRecord();
       String[] hostnameSplit = helixInstanceConfig.getHostName().split("_");
       Preconditions.checkState(hostnameSplit.length >= 2);
-      String port = record.getSimpleField(CommonConstants.Helix.Instance.ADMIN_PORT_KEY);
-      endpointToInstance.put(instance, hostnameSplit[1] + ":" + port);
+      String adminPort = record.getSimpleField(CommonConstants.Helix.Instance.ADMIN_PORT_KEY);
+      if (Strings.isNullOrEmpty(adminPort)) {
+        LOGGER.warn("Admin port is missing for host: {}. Using the default port: {}",
+            hostnameSplit[1], CommonConstants.Server.DEFAULT_ADMIN_API_PORT);
+        adminPort = Integer.toString(CommonConstants.Server.DEFAULT_ADMIN_API_PORT);
```

Comment:
You should not be importing Server constants into controller. Please remove the code to pick up default admin port, because the server updates its own instance config with the admin port information. If somehow that update is wrong, then we should not try connecting to that server at all. Please be sure to print helixInstanceConfig.getHostName() whenever the admin port is not present or not parsable.